hyperspy/models/edsmodel.py

```diff
@@ -317,6 +317,56 @@ def disable_xray_lines(self):
         """
         for component in self._active_xray_lines:
             component.active = False
+    def fit(self, fitter='mpfit', method='ls', grad=False,
```

Comment:
If all the components of the model have components with the gradients defined (`Gaussian` does), we may consider setting `grad=True` as this should speed things up.